discovery-infra/start_discovery.py

```diff
@@ -436,7 +451,11 @@ def set_hosts_roles(client, cluster, nodes_details, machine_net, tf, master_coun
     if machine_net.has_ip_v4:
         libvirt_nodes = utils.get_libvirt_nodes_mac_role_ip_and_name(networks_names[0])
         libvirt_nodes.update(utils.get_libvirt_nodes_mac_role_ip_and_name(networks_names[1]))
-        update_hostnames = False
+        if static_ip_mode:
+            log.info("Setting hostnames when running in static ips mode")
+            update_hostnames = True
+        else:
+            update_hostnames = True
```

Comment:
Looks like a bug. You're setting `update_hostname = True` in both cases.

Reply:
yes, i will fix it as part of MGMT-3568